server/fleet/errors.go

```diff
@@ -19,6 +19,13 @@ var (
 	ErrMDMNotConfigured      = &MDMNotConfiguredError{}
 )
 
+// ErrWithStatusCode is an interface for errors that should set a specific HTTP
+// status code.
+type ErrWithStatusCode interface {
+	error
+	StatusCode() int
+}
```

Comment:
:+1: that's a good idea, up until now we've used `kithttp.StatusCoder` but it's not ideal to encode that interface in a third-party package.